CoffeeMachine.java

```diff
@@ -0,0 +1,150 @@
+                System.out.println("Your coffee is done");
+                this.espressoSingleShot++;
+                System.out.println("You made "+ this.espressoSingleShot+ " single shots so far!.");
+            } else {
+                System.out.println("Not the right amount of coffee grounds");
+            }
+    }
+private boolean makeEspressoSingleShot(int coffeeGrounds) throws InterruptedException {
+if(coffeeGrounds>=6 && coffeeGrounds<=8){
+    System.out.print("Brewing time(WAIT):");
+    Thread.sleep(20000);
+    return true;
+}
+else return false;
+}
+public void DoubleShotDoppio() throws InterruptedException {
+    System.out.println("Would you like to add cold water(y/n)?");
+    char answer=object.next().charAt(0);
+    if(addWater(answer)){
+        this.coldWater=5;
+        System.out.println("Added "+ this.coldWater+ " ounces of cold water");
+    }
+    System.out.print("Put coffeeGrounds:");
+    this.coffeeGrounds = object.nextInt();
+    if (makeDoubleShotDoppio(this.coffeeGrounds)) {
+        System.out.println("Your coffee is done");
+        this.DoubleShotDoppio++;
+        System.out.println("You made "+ this.DoubleShotDoppio+ " double shots so far!");
+    } else {
+        System.out.println("Not the right amount of coffee grounds");
+    }
+}
+    private boolean makeDoubleShotDoppio(int coffeeGrounds) throws InterruptedException {
+        if(coffeeGrounds>=18 && coffeeGrounds<=21){
+            System.out.print("Brewing time(WAIT):");
+            Thread.sleep(30000);
+            return true;
+        }
+        else return false;
+    }
+    public void Lungo() throws InterruptedException {
+        System.out.println("Would you like to add cold water(y/n)?");
+        char answer=object.next().charAt(0);
+        if(addWater(answer)){
+            this.coldWater=6;
+            System.out.println("Added "+ this.coldWater+ " ounces of cold water");
+        }
+        System.out.print("Put coffeeGrounds:");
+        this.coffeeGrounds = object.nextInt();
+        if (makeLungo(this.coffeeGrounds)) {
+            System.out.println("Your coffee is done");
+            this.Lungo++;
+            System.out.println("You made "+ this.Lungo+ " lungo so far!.");
+        } else {
+            System.out.println("Not the right amount of coffee grounds");
+        }
+    }
+    private boolean makeLungo(int coffeeGrounds) throws InterruptedException {
+        if(coffeeGrounds>=18 && coffeeGrounds<=21){
+            System.out.print("Brewing time(WAIT):");
+            Thread.sleep(40000);
+            return true;
+        }
+        else return false;
+    }
+    public void Macchiato() throws InterruptedException {
+        if(addSteamedMilk()){
+            System.out.println("Milk is  steamed!");
+        }
+        System.out.print("Put coffeeGrounds:");
+        this.coffeeGrounds = object.nextInt();
+        if (makeLungo(this.coffeeGrounds)) {  //ista funkcija jer treba ista kolicina vremena

```

Comment:
You can have some general method name which is not lungo or macchiato if it is used for the same purpose. 
Alternatively and even better, you can have general method for making of any coffee type which takes as parameters all variations needed for making any specific coffeee.